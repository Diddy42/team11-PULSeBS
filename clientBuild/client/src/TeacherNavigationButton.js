import React from 'react';
import {Link} from 'react-router-dom';

class TeacherNavigationButton extends React.Component{
    constructor(props){
        super(props);

        this.state = { onCalendar: true };
    }

    render(){
        return <>
            {
                this.state.onCalendar &&
                <Link className='mr-4' to='/teacher/historicalData' onClick={this.handleClick}>
                    Historical data page
                </Link>
            }

            {
                !this.state.onCalendar &&
                <Link className='mr-4' to='/teacher' onClick={this.handleClick}>
                    Calendar page
                </Link>
            }
        </>
    }

    handleClick = () => {
        this.setState({ onCalendar: !this.state.onCalendar });
    }
}

export default TeacherNavigationButton;