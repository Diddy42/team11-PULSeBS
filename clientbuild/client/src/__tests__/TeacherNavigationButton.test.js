import TestRenderer from 'react-test-renderer';
import React from 'react';
import TeacherNavigationButton from '../TeacherNavigationButton';
import { shallow } from 'enzyme';
import {BrowserRouter as Router} from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('renders learn react link', () => {
    const tree = TestRenderer.create(<Router><TeacherNavigationButton/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });


test('handleClick', async () => {
  const component = shallow(<TeacherNavigationButton/>);
  const instance = component.instance();
  expect(instance.state.onCalendar).toBe(true);
  await instance.handleClick();
  expect(instance.state.onCalendar).toBe(false);
});