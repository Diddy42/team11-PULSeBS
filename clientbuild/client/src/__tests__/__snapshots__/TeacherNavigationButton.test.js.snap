// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button 1`] = `
<a
  className="mr-4 text-light"
  href="/teacher/historicalData"
  onClick={[Function]}
>
  Historical data page
</a>
`;
